### PR TITLE
Matter eater now moves items to the chest, fixes some hidden item bugs

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Instead of deleting items eaten by matter eater, moves them to the eater's chest organ. These items can be removed with cavity surgery, or by disemboweling the eater.

Also, fixes some bugs with cavity surgery, and improves how matter eater handles things on your person now that they're not qdeleted.

## Why It's Good For The Game
This doesn't make matter eater more containable, but at least means you can undo the damage if a random geneticist eats the armory.

## Testing
Ate some stuff and then cut my stomach open, spilling out the things onto the floor

## Changelog
:cl:
tweak: Matter eater now moves items to your chest instead of deleting them.
/:cl:

### CODE REVIEW

File: code/game/dna/mutations/mutation_powers.dm [hunk 1]
```diff
@@ -446,9 +446,22 @@
 	else
 		user.visible_message("<span class='danger'>[user] eats \the [the_item].</span>")
 		playsound(user.loc, 'sound/items/eatfood.ogg', 50, 0)
+		if(ismob(the_item.loc) && isitem(the_item))
```
Comment (1080pCat) [inline code/game/dna/mutations/mutation_powers.dm:449]: `isitem(the_item)`
dies

File: code/game/dna/mutations/mutation_powers.dm [hunk 2]
```diff
@@ -446,9 +446,22 @@
...
+			var/obj/item/organ/external/chest/target_place = H.get_organ(BODY_ZONE_CHEST)
+			if(istype(target_place))
+				the_item.forceMove(target_place)
+				doHeal(user)
+				return
+
 		qdel(the_item)
 		doHeal(user)
 
+
```
Comment (1080pCat) [inline code/game/dna/mutations/mutation_powers.dm:464]:
```suggestion

```
double white space? why?

File: code/modules/surgery/organs/organ_external.dm [hunk 3]
```diff
@@ -645,8 +646,17 @@ Note that amputating the affected organ does in fact remove the infection from t
 			O.forceMove(T)
 			organ_spilled = TRUE
 
+	var/obj/item/organ/external/org = C.get_organ(spillage_zone)
+	if(istype(org))
+		for(var/obj/item/content in org)
+			content.forceMove(T)
+			content.dropped(owner, TRUE)
+
+
```
Comment (1080pCat) [inline code/modules/surgery/organs/organ_external.dm:655]:
```suggestion

```
same here

**Review by 1080pCat — COMMENTED** [review]
Lets say im a plasmaman, and i eat the nuke core, will i continue to be a radiation hazard?

File: code/game/dna/mutations/mutation_powers.dm (hunk 2 above)
Comment (lewcc) [reply code/game/dna/mutations/mutation_powers.dm:464]: forgor

File: code/game/dna/mutations/mutation_powers.dm [hunk 4]
```diff
@@ -446,6 +446,18 @@
 	else
 		user.visible_message("<span class='danger'>[user] eats \the [the_item].</span>")
 		playsound(user.loc, 'sound/items/eatfood.ogg', 50, 0)
```
Comment (Burzah) [inline code/game/dna/mutations/mutation_powers.dm:448]:
```suggestion
		playsound(user.loc, 'sound/items/eatfood.ogg', 50, FALSE)
```

File: code/modules/surgery/organs/organ_external.dm [hunk 5]
```diff
@@ -634,6 +634,7 @@ Note that amputating the affected organ does in fact remove the infection from t
 		return
 
 	var/organ_spilled = FALSE
+	var/something_else = FALSE
 	var/turf/T = get_turf(C)
 	C.add_splatter_floor(T)
 	playsound(get_turf(C), 'sound/effects/splat.ogg', 25, 1)
```
Comment (Burzah) [inline code/modules/surgery/organs/organ_external.dm:640]:
```suggestion
	playsound(get_turf(C), 'sound/effects/splat.ogg', 25, TRUE)
```

File: code/modules/surgery/organs/organ_external.dm [hunk 6]
```diff
@@ -645,8 +646,16 @@ Note that amputating the affected organ does in fact remove the infection from t
...
 			organ_spilled = TRUE
 
+	var/obj/item/organ/external/org = C.get_organ(spillage_zone)
+	if(istype(org))
+		for(var/obj/item/content in org)
+			content.forceMove(T)
+			content.dropped(owner, TRUE)
+
 	if(organ_spilled)
 		C.visible_message("<span class='danger'><B>[C]'s internal organs spill out onto the floor!</B></span>")
```
Comment (Burzah) [inline code/modules/surgery/organs/organ_external.dm:656]:
```suggestion
		C.visible_message("<span class='danger'><b>[C]'s internal organs spill out onto the floor!</b></span>")
```

**Review by Burzah — COMMENTED** [review]
I know you didn't touch this stuff, but while we are here.

File: code/game/dna/mutations/mutation_powers.dm [hunk 7]
```diff
@@ -445,7 +445,19 @@
 			doHeal(user)
 	else
 		user.visible_message("<span class='danger'>[user] eats \the [the_item].</span>")
```
Comment (DGamerL) [inline code/game/dna/mutations/mutation_powers.dm:447]:
```suggestion
	else
		user.visible_message("<span class='danger'>[user] eats [the_item].</span>")
```

File: code/game/dna/mutations/mutation_powers.dm [hunk 8]
```diff
@@ -445,7 +445,19 @@
...
-		playsound(user.loc, 'sound/items/eatfood.ogg', 50, 0)
+		playsound(user.loc, 'sound/items/eatfood.ogg', 50, FALSE)
+		if(ismob(the_item.loc) && isitem(the_item))
+			var/obj/item/eaten = the_item
+			var/mob/the_owner = the_item.loc
+			the_owner.unEquip(eaten, TRUE, TRUE)
```
Comment (DGamerL) [inline code/game/dna/mutations/mutation_powers.dm:452]: Shouldn't this rather check if it's unequippable even?

File: code/modules/surgery/organs/organ_external.dm [hunk 9]
```diff
@@ -634,9 +634,10 @@ Note that amputating the affected organ does in fact remove the infection from t
 		return
 
 	var/organ_spilled = FALSE
+	var/something_else = FALSE
```
Comment (DGamerL) [inline code/modules/surgery/organs/organ_external.dm:637]: Do we ever set this to `TRUE`?

**Review by DGamerL — CHANGES_REQUESTED** [review]

**Comment by lewcc** [comment]
> > Lets say im a plasmaman, and i eat the nuke core,
> 
> sadly you probably can't eat high-value objects, if you ate something very radioactive though maybe?

You would think that. You would certainly think that

**Comment by Rayforx** [comment]
> > > Lets say im a plasmaman, and i eat the nuke core,
> > 
> > 
> > sadly you probably can't eat high-value objects, if you ate something very radioactive though maybe?
> 
> You would think that. You would certainly think that

Funnily enough, there is no check for matter eater on any of the high-risk items.

Yes, you can even eat the nuke disk, and it teleports somewhere else, so you will probably end up with a matter eater to protect the disk if it's stored and not destroyed anymore.

File: code/game/dna/mutations/mutation_powers.dm (hunk 8 above)
Comment (lewcc) [reply code/game/dna/mutations/mutation_powers.dm]: Done

File: code/modules/surgery/organs/organ_external.dm (hunk 9 above)
Comment (lewcc) [reply code/modules/surgery/organs/organ_external.dm:637]: oops, good catch. fixed

Comment (Sirryan2002) [inline code/modules/surgery/organs/organ_external.dm:637]: could you leave a comment explaining what this var is tracking for the uninformed

File: code/game/dna/mutations/mutation_powers.dm [hunk 10]
```diff
@@ -444,8 +444,23 @@
...
+
+		if(ismob(the_item.loc) && isitem(the_item))
+			var/obj/item/eaten = the_item
+			var/mob/the_owner = the_item.loc
+			if(!the_owner.unEquip(eaten, FALSE, TRUE))
+				to_chat(user, "<span class='warning'>You can't eat [the_item], it won't go down your throat!</span>")
+				return
+		user.visible_message("<span class='danger'>[user] eats [the_item].</span>")
+		playsound(user.loc, 'sound/items/eatfood.ogg', 50, FALSE)
+		if(ishuman(user))
```
[the remark this answers is not in the document]
Comment (lewcc) [reply code/game/dna/mutations/mutation_powers.dm:456]: Slightly, it's important that it falls through to the qdel n stuff

**Review by DGamerL — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Pull request merged**